spec.bs

```diff
@@ -136,3 +136,115 @@ dictionary URLPatternComponentResult {
   record<USVString, USVString> groups;
 };
 </xmp>
+
+Each {{URLPattern}} object has an associated <dfn for=URLPattern>protocol component</dfn>, a <a for=/>component</a>, initially null.
+
+Each {{URLPattern}} object has an associated <dfn for=URLPattern>username component</dfn>, a <a for=/>component</a>, initially null.
+
+Each {{URLPattern}} object has an associated <dfn for=URLPattern>password component</dfn>, a <a for=/>component</a>, initially null.
+
+Each {{URLPattern}} object has an associated <dfn for=URLPattern>hostname component</dfn>, a <a for=/>component</a>, initially null.
+
+Each {{URLPattern}} object has an associated <dfn for=URLPattern>port component</dfn>, a <a for=/>component</a>, initially null.
+
+Each {{URLPattern}} object has an associated <dfn for=URLPattern>pathname component</dfn>, a <a for=/>component</a>, initially null.
+
+Each {{URLPattern}} object has an associated <dfn for=URLPattern>search component</dfn>, a <a for=/>component</a>, initially null.
+
+Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component</dfn>, a <a for=/>component</a>, initially null.
+
+<dl class="domintro non-normative">
+  <dt><code>{{URLPattern}} . {{URLPattern/protocol}}</code>
+  <dd>
+    <p>The normalized protocol pattern string.
```

Comment:
Web developers might enjoy these being cross-linked to the appropriate parts of the URL spec, e.g. `The normalized [=URL/scheme=] pattern string`. But I'm not sure.

Reply:
Hmm, I tried this but its failing with "LINK ERROR: No 'dfn' refs found for 'scheme' with for='['URL']'."  Do I need to include it somehow?

Reply:
Oh, it appears to need lowercase `url/scheme`

Reply:
Hmm, I think I'm not going to do this.  The names don't quite match up (scheme-vs-protocol, search-vs-query, etc).  Also, if you then follow the links they give you definitions for how they should be defined that are not true for pattern strings.

Reply:
Yeah, the latter was my main uncertainty. Makes sense.